File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/strategy/transport/TransportStrategy.java
package com.bigboxcode.designpattern.strategy.transport;

public class TransportStrategy {
    private final Transport transport;

    public TransportStrategy(Transport transport) {
        this.transport = transport;
    }

    public void execute() {
        this.transport.start();

        this.transport.getInfo();

        this.transport.operate();
    }

    public void repair() {
        this.transport.repair();
    }

    public void stop() {
        this.transport.stop();
    }
}

File: design-patterns/ts/strategy/transport/transport-strategy.ts
// transport-strrategy.ts

import Transport from "./transport";

class TransportStrategy {
    private transport: Transport;

    constructor(transport: Transport) {
        this.transport = transport;
    }

    execute(): void {
        this.transport.start();

        this.transport.getInfo();

        this.transport.operate();
    }

    repair(): void {
        this.transport.repair();
    }

    stop(): void {
        this.transport.stop();
    }
}

export default TransportStrategy;